src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalUsingVarRewriter.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.Lowering.LocalRewriter
+{
+    class LocalUsingVarRewriter : BoundTreeRewriterWithStackGuard
+    {
+        public static BoundNode Rewrite(BoundStatement statement)
+        {
+            var localUsingVarRewriter = new LocalUsingVarRewriter();
+            return (BoundNode)localUsingVarRewriter.Visit(statement);
+        }
+
+        public override BoundNode VisitBlock(BoundBlock node)
+        {
+            int current = 0;
+            ImmutableArray<BoundStatement> statements = (ImmutableArray<BoundStatement>)this.VisitList(node.Statements);
+
+            foreach (BoundStatement statement in statements)
+            {
+                if (statement is BoundLocalDeclaration boundAssignment)
```

Comment:
>boundAssignment [](start = 55, length = 15)

Consider renaming since this is not simply an assignment. Perhaps `localDeclaration`.